tests/providers/cncf/kubernetes/utils/test_pod_manager.py

```diff
@@ -346,6 +351,24 @@ def test_pod_manager_get_client_call_deprecation(self):
                 "and make kube_client a required argument."
             )
 
+    @pytest.mark.parametrize(
+        "container_state, is_terminated", [("waiting", False), ("running", False), ("terminated", True)]
+    )
+    def test_container_is_terminated_with_waiting_state(self, container_state, is_terminated):
+        container_status = MagicMock()
+        container_status.configure_mock(
+            **{
+                "name": "base",
+                "state.waiting": True if container_state == "waiting" else None,
+                "state.running": True if container_state == "running" else None,
+                "state.terminated": True if container_state == "terminated" else None,
+            }
+        )
+        pod_info = MagicMock()
+        pod_info.status.container_statuses = [container_status]
+        is_terminated = container_is_terminated(pod_info, "base")
+        assert is_terminated == is_terminated
```

Comment:
```suggestion
        assert container_is_terminated(pod_info, "base") == is_terminated
```

You are overwriting the `is_terminated` you are parameterizing! (I personally like using `expected_` for expectations that are parameterized, but feel free to do it however you'd like)

Reply:
My bad, I just fixed it.